migrations/20230113201828-create-tag-stats-materialized-view.ts

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`
+    CREATE MATERIALIZED VIEW IF NOT EXISTS "CollectiveTagStats" AS
+    SELECT DISTINCT unnest(tags) AS tag, COUNT(*) AS count FROM "Collectives" GROUP BY tag ORDER BY count DESC
```

Comment:
I'm afraid of the pollution that would come from non-legit accounts. We should at least add a `WHERE "deletedAt" IS NULL`. I would personally go a bit further than that:
- I don't think we care about tags on user profiles
- If we do care about them, we should at least filter `isIncognito IS NOT TRUE AND (c."data" ->> 'isGuest')::boolean IS NOT TRUE `

To prevent SPAM, we may also want to pick some of the conditions implemented for search, such as `skipRecentAccounts` (`AND (COALESCE((c."data"#>>'{spamReport,score}')::float, 0) <= 0.2 OR c."createdAt" < (NOW() - interval '2 day'))`).

Also `AND ((data ->> 'hideFromSearch'::text)::boolean) IS NOT TRUE`

Reply:
@Betree great feedback! 

Regarding user profile tags, I agree that user tags might not be too interesting. However:
  - We have the edit tags field on User collectives in the settings, although probably not very well used since we don't show tags on User profile pages, so maybe a "suboptimal experience" there could be ok by either:
    - skipping autocompletion on that field
    - just autocomplete on Collective tags
- the current query is including User tags, but I don't think it is really too interesting there either since the query is currently only used on the search page which does not include User collectives. So we might want to filter out User collectives in that query too to keep them consistent, if we filter out User tags in this view.

Hmm. I'm happy to go either way, what do you think?

Reply:
No strong opinion on my side; I would say go with the solution that seems to be the simplest for you

Reply:
@Betree Ok, I'll keep the inclusion of User tags as is for now, will probably require less changes overall. Just pushed the fixes you suggested